src/lib/Normalize.hs

```diff
@@ -96,6 +125,20 @@ normalizeTLam (FTLam tvs qs body) =
     body' <- extendR env $ normalize body
     return (qs', body')
 
+desugarType :: Type -> Type
+desugarType (SumTy l r) = RecTy $ Tup $ [l, r, TC $ BaseType BoolType]
+desugarType tv@(TypeVar _) = tv
+desugarType (ArrowType l (Pi a (eff, b))) = ArrowType l $ Pi (desugarType a) (eff, desugarType b)
+desugarType (TabType (Pi a b)) = TabType $ Pi (desugarType a) (desugarType b)
+desugarType (Forall vs qs t)   = Forall vs qs $ desugarType t
+desugarType (TypeAlias vs t)   = TypeAlias vs $ desugarType t
+desugarType (TC con)           = TC $ fmapTyCon con desugarType id
+desugarType eff@(Effect _ _)   = eff
+desugarType NoAnn = NoAnn
```

Comment:
I really don't like the fact that it's a poor man's traversal, but making `Type` a functor is annoying, because you'd need to take a fix-point and that introduces more constructors to unpack 🤮

Reply:
Without changing `Type`, we could make a type-blind traversal like `traverseType :: Applicative f => Type -> (Type -> f Type) -> f Type`. But this looks fine to me. Most of the cases are still handled by `fmapTyCon`.